bpl: read wireless configuration by interface name instead of index

adam1985d opened the pull request:
This PR breaks the dependency of BPL in wireless radio section naming in the UCI config.
The new implementation looks-up the interface name in the wireless config.
If the param is requested from the main radio - a search of the param value is done in the `wireless.<device>.param` where <device> is the device name found in the section where the interface name was found.
```
e.g.
lookup **great_iface_name**:
wireless.**adam**.ifname=**great_iface_name**->
wireless.**adam**.device=**adam_device** ->
read: wireless.**adam_device**.<param>
```
for VAPs, the param is searched in the current section, but again - the section is located using the interface name:
```
lookup **great_iface_name**:
wireless.**adam**.ifname=**great_iface_name**->
read: wireless.**adam**.<param>
```

this PR is rebased to the following PR:
https://github.com/prplfoundation/prplMesh/pull/785

Fixes #574

vitalybu commented on common/beerocks/bcl/include/bcl/beerocks_async_work_queue.h:
```diff
@@ -20,7 +20,7 @@ namespace beerocks {
 class async_work_queue : public thread_base {
 public:
-    async_work_queue()  = default;
+    async_work_queue() = default;
```
All non feed related files should be removed from this commit, as they are clang related issues, and were already fixed in the master.

adam1985d replied:
I will rebase again to master

vitalybu commented on agent/src/beerocks/slave/beerocks_slave_main.cpp:
```diff
@@ -498,6 +498,26 @@ int main(int argc, char *argv[])
+    }
+
+    //create unordered_map of interfaces
+    std::unordered_map<int, std::string> interfaces_map;
```
If the map is empty at the end of this loop, maybe there's no reason to continue...

adam1985d replied:
would you consider this case as error or return 0?

vitalybu replied:
Log an INFO and return 0

adam1985d replied:
fixed

vitalybu commented on agent/src/beerocks/slave/beerocks_slave_main.cpp:
```diff
@@ -182,26 +182,35 @@ static void fill_son_slave_config(beerocks::config_file::sConfigSlave &beerocks_
+        if (!beerocks::os_utils::is_pid_running(temp_path, base_name)) {
+            //start new slave process
+            std::string file_name = "./" + std::string(BEEROCKS_AGENT);
+            if (access(file_name.c_str(), F_OK) == -1) //file does not exist in current location
```
```suggestion
            if (access(file_name.c_str(), F_OK) == -1) { //file does not exist in current location
```

adam1985d replied:
fixed

vitalybu commented on agent/src/beerocks/slave/beerocks_slave_main.cpp:
```diff
@@ -182,26 +182,35 @@ static void fill_son_slave_config(beerocks::config_file::sConfigSlave &beerocks_
+            //start new slave process
+            std::string file_name = "./" + std::string(BEEROCKS_AGENT);
+            if (access(file_name.c_str(), F_OK) == -1) //file does not exist in current location
+            {
```
```suggestion
```

adam1985d replied:
fixed

vitalybu commented on agent/src/beerocks/slave/beerocks_slave_main.cpp:
```diff
@@ -552,24 +550,31 @@ int main(int argc, char *argv[])
+        LOG(INFO) << "Starting slave for iface '" << hostap_iface << "'";
+
+        std::string file_name = "./" + std::string(BEEROCKS_AGENT);
+        if (access(file_name.c_str(), F_OK) == -1) //file does not exist in current location
```
```suggestion
        if (access(file_name.c_str(), F_OK) == -1) { //file does not exist in current location
```

adam1985d replied:
fixed

vitalybu commented on agent/src/beerocks/slave/beerocks_slave_main.cpp:
```diff
@@ -552,24 +550,31 @@ int main(int argc, char *argv[])
+
+        std::string file_name = "./" + std::string(BEEROCKS_AGENT);
+        if (access(file_name.c_str(), F_OK) == -1) //file does not exist in current location
+        {
```
```suggestion
```

adam1985d replied:
fixed

vitalybu commented on framework/platform/bpl/include/bpl/bpl_cfg.h:
```diff
@@ -214,15 +214,6 @@ struct BPL_WLAN_PARAMS {
 
     /* Wi-Fi Channel (0 for ACS) */
     int channel;
-
-    /* Wi-Fi SSID */
-    char ssid[BPL_SSID_LEN];
-
-    /* Wi-Fi KeyPassphrase */
-    char passphrase[BPL_PASS_LEN];
-
-    /* Wi-Fi Securirt Mode */
-    char security[BPL_SEC_LEN];
```
Are you sure it's the right thing to do? We won't need it in the future?
I would leave it for now anyway, as it has nothing to do with the current PR.

adam1985d replied:
this information isn't used at all - although read periodically.
it is checked for change and the agent is alerted, but the agent ignores this information and only handles a change in `band_enabled`.

there is a task to remove the rest of the settings read from wireless config from there and read using NL message or similar.

vitalybu replied:
Reading this information in a different way is fine, but still it feels like this change should not be part of this PR.

tomereli replied:
As I recall, there was another reason apart from the fact this is not used which is why @adam1985d removed these - perhaps it should be explained in the commit message.

adam1985d replied:
the reason to remove was that we are not using the code.
the reason why we were looking at this at all was that due to the difference in the wireless configuration between UGW, GLInet, RAX40, etc.

we could have changed those wifi settings to optional - but that would just hide the fact we aren't really reading it.
we could have invested in fixing the wireless configuration in GLInet and Turris Omnia - but the decision was not to invest in this at this point for several reasons:
1. the data read is not really used by the prplmesh
2. there is a task to remove the prplmesh access to wireless configuration and instead use BWL or some other method to retrieve the information.

regarding the commit message, I don't think that the platform wireless configuration difference is really beneficial information here since if we wanted to address it we would have fixed it.

vitalybu commented on agent/src/beerocks/slave/platform_manager/platform_manager_thread.cpp:
```diff
@@ -99,16 +99,9 @@ static bool fill_platform_settings(
     /* update message */
     msg->wlan_settings().band_enabled = params.enabled;
     msg->wlan_settings().channel      = params.channel;
-    string_utils::copy_string(msg->wlan_settings().ssid, params.ssid,
```
Some as the previous commit.
I'm not sure this information won't be used in the future, and it's not part of the current PR anyway.
Should be reverted.

tomereli replied:
Since currently it is not used I don't think there's a problem removing it.
Also I don't see why prplmesh should ever need to read the SSID from BPL. Same for the other deleted configuration parameters.
@arnout what do you think?

vitalybu commented on framework/platform/bpl/uci/cfg/bpl_cfg_uci.cpp:
```diff
@@ -162,6 +162,128 @@ int cfg_uci_get_wireless(enum paramType type, int index, const char param[], cha
+    struct uci_element *e;
+    struct uci_section *s;
+    struct uci_element *n;
+    char lookup_str[MAX_UCI_BUF_LEN] = "wireless";
```
We're in a C++ world, why not define the variables where you need them?
Especially if you name them as `p`, `e`, `s` & `n`...

adam1985d replied:
fixed

vitalybu commented on framework/platform/bpl/uci/cfg/bpl_cfg_uci.cpp:
```diff
@@ -162,6 +162,128 @@ int cfg_uci_get_wireless(enum paramType type, int index, const char param[], cha
+
+    ctx = uci_alloc_context();
+    if (!ctx)
+        return RETURN_ERR;
```
For example, I would define the `ctx` here: `uci_context *ctx = uci_alloc_context();`
But if you don't, you can at least allocate and validate on the same line: `if (!(ctx = uci_alloc_context())) {`
Lastly, always surround `if` statements with `{ }`, even one-liners, as at some point someone can replace your code with a multi-line macro...

adam1985d replied:
fixed

vitalybu commented on framework/platform/bpl/uci/cfg/bpl_cfg_uci.cpp:
```diff
@@ -162,6 +162,128 @@ int cfg_uci_get_wireless(enum paramType type, int index, const char param[], cha
+    if (!ctx)
+        return RETURN_ERR;
+
+    if ((uci_lookup_ptr(ctx, &ptr, lookup_str, true) != UCI_OK)) {
```
Here I would define the `ptr` and `lookup_str` variables:
```suggestion
    struct uci_ptr ptr;
    const char lookup_str[MAX_UCI_BUF_LEN] = "wireless";
    if ((uci_lookup_ptr(ctx, &ptr, lookup_str, true) != UCI_OK)) {
```

adam1985d replied:
fixed

vitalybu commented on framework/platform/bpl/uci/cfg/bpl_cfg_uci.cpp:
```diff
@@ -162,6 +162,128 @@ int cfg_uci_get_wireless(enum paramType type, int index, const char param[], cha
+        return RETURN_ERR;
+    }
+
+    p = ptr.p;
```
Probably should make sure that `ptr` and `ptr.p` are valid.
```C++
if (!ptr || !ptr.p) { ... }
```

adam1985d replied:
fixed

vitalybu commented on framework/platform/bpl/uci/cfg/bpl_cfg_uci.cpp:
```diff
@@ -162,6 +162,128 @@ int cfg_uci_get_wireless(enum paramType type, int index, const char param[], cha
+
+    p = ptr.p;
+
+    uci_foreach_element(&p->sections, e)
```
This entire code block requires some commenting.
For example - `// Iterate over all "wireless." sections in the UCI database`

adam1985d replied:
added more comments

vitalybu commented on framework/platform/bpl/uci/cfg/bpl_cfg_uci.cpp:
```diff
@@ -162,6 +162,128 @@ int cfg_uci_get_wireless(enum paramType type, int index, const char param[], cha
+        {
+            struct uci_option *o = uci_to_option(n);
+
+            if (strncmp(n->name, "device", MAX_UCI_BUF_LEN) == 0 && o->type == UCI_TYPE_STRING) {
```
Better check the type first, for two reasons:
1. Native type comparison is far more efficient than string comparison.
2. If the type is not `UCI_TYPE_STRING`, your code will `strncmp()` some unknown type.

adam1985d replied:
ok - changed the order of checks

vitalybu commented on framework/platform/bpl/uci/cfg/bpl_cfg_uci.cpp:
```diff
@@ -162,6 +162,128 @@ int cfg_uci_get_wireless(enum paramType type, int index, const char param[], cha
+        deviceName[0] = '\0';
+        uci_foreach_element(&s->options, n)
+        {
+            struct uci_option *o = uci_to_option(n);
```
Can `o` be `NULL`?

adam1985d replied:
it is an offset calculation back inside the current line, so I think no.

vitalybu commented on framework/platform/bpl/uci/cfg/bpl_cfg_uci.cpp:
```diff
@@ -162,6 +162,128 @@ int cfg_uci_get_wireless(enum paramType type, int index, const char param[], cha
+    if (type == TYPE_RADIO) {
+        //get radio name from .device param value
+        char deviceName[MAX_UCI_BUF_LEN];
+        deviceName[0] = '\0';
```
Why not initialize the entire array to `{}` or `{0}` on definition (one line above)?

adam1985d replied:
fixed

vitalybu commented on framework/platform/bpl/uci/cfg/bpl_cfg.cpp:
```diff
@@ -157,33 +157,30 @@ int cfg_get_device_info(BPL_DEVICE_INFO *device_info) { return 0; }
-    int index  = 0;
-
     if (!iface || !wlan_params) {
+        MAPF_ERR("cfg_get_wifi_params: invalid input: iface=" << intptr_t(iface) << " wlan_params="
```
```suggestion
        MAPF_ERR("cfg_get_wifi_params: invalid input: iface = " << intptr_t(iface) << " wlan_params = "
```

adam1985d replied:
fixed

vitalybu commented on framework/platform/bpl/uci/cfg/bpl_cfg.cpp:
```diff
@@ -157,33 +157,30 @@ int cfg_get_device_info(BPL_DEVICE_INFO *device_info) { return 0; }
-
     if (!iface || !wlan_params) {
+        MAPF_ERR("cfg_get_wifi_params: invalid input: iface=" << intptr_t(iface) << " wlan_params="
+                                                              << intptr_t(wlan_params) << "\n");
```
```suggestion
                                                              << intptr_t(wlan_params));
```

adam1985d replied:
fixed

vitalybu commented on framework/platform/bpl/uci/cfg/bpl_cfg_uci.cpp:
```diff
@@ -162,6 +162,132 @@ int cfg_uci_get_wireless(enum paramType type, int index, const char param[], cha
+        return RETURN_ERR;
+    }
+
+    if (!ptr.p) {
```
`ptr` should be initialized to `nullptr` and checked for validity here (mentioned this in one of the previous comments)

adam1985d replied:
the naming here is a bit confusing.
although named ptr, it is actually a statically defined struct with set of pointers and other information on the data:
```
struct uci_ptr
{
	enum uci_type target;
	enum {
		UCI_LOOKUP_DONE =     (1 << 0),
		UCI_LOOKUP_COMPLETE = (1 << 1),
		UCI_LOOKUP_EXTENDED = (1 << 2),
	} flags;

	struct uci_package *p;
	struct uci_section *s;
	struct uci_option *o;
	struct uci_element *last;

	const char *package;
	const char *section;
	const char *option;
	const char *value;
};
```

tomereli commented on framework/platform/bpl/uci/cfg/bpl_cfg_uci.cpp:
```diff
@@ -162,6 +162,132 @@ int cfg_uci_get_wireless(enum paramType type, int index, const char param[], cha
     return status;
 }
 
+int cfg_uci_get_wireless_from_ifname(enum paramType type, char *interface_name, const char param[],
```
In general, the commit message should not use the word "break" unless its doing something bad :)
I think a better way to phrase this is "This API removes the unnecessary dependency ..."

adam1985d replied:
fixed

tomereli commented on framework/platform/bpl/uci/cfg/bpl_cfg_uci.cpp:
```diff
@@ -162,6 +162,132 @@ int cfg_uci_get_wireless(enum paramType type, int index, const char param[], cha
+    }
+
+    char lookup_str[MAX_UCI_BUF_LEN] = "wireless";
+    struct uci_ptr ptr;
```
I don't think there is a need to use the `struct` keyword since this is CPP (nitpick)

arnout replied:
I think it's needed because the type is declared within an `extern "C"`.

tomereli commented on framework/platform/bpl/uci/cfg/bpl_cfg_uci.cpp:
```diff
@@ -162,6 +162,132 @@ int cfg_uci_get_wireless(enum paramType type, int index, const char param[], cha
+
+        status = cfg_uci_get(path, value, MAX_UCI_BUF_LEN);
+        if (status == RETURN_OK) {
+            // DEBUG("%s index=%d %s=%s\n", __func__, index, param, value);
```
Removed dead code

adam1985d replied:
fixed

tomereli commented on framework/platform/bpl/uci/cfg/bpl_cfg.cpp:
```diff
@@ -175,33 +175,30 @@ int cfg_get_device_info(BPL_DEVICE_INFO *device_info) { return 0; }
     // The UCI "disabled" setting is optional, defaults to false if not present
     bool disabled = false;
-    cfg_uci_get_wireless_bool(TYPE_RADIO, index, "disabled", &disabled);
+    cfg_uci_get_wireless_bool(TYPE_RADIO, ifname, "disabled", &disabled);
```
Not OK - this can fail, I know this wasn't tested before but it should definitely be tested now since `ifname` is not guaranteed to be in `/etc/config/wireless`.

arnout replied:
Well... If it fails, it can be because the section is missing (a real error), or because the option is not set in UCI (default false, as explained in the comment above). So this *is* actually OK IMO.

tomereli replied:
Right, I get it.

tomereli commented on framework/platform/bpl/uci/cfg/bpl_cfg.cpp:
```diff
@@ -175,33 +175,30 @@ int cfg_get_device_info(BPL_DEVICE_INFO *device_info) { return 0; }
     }
 
-    return retVal;
+    return RETURN_OK;
```
We should not always return success here - depending on the actual operation success/failure.

arnout replied:
... but it was already like that, so it should be fixed in a separate commit.

And as I explained, I think it actually is OK.

tomereli commented on framework/platform/bpl/uci/cfg/bpl_cfg_helper.cpp:
```diff
@@ -82,14 +82,17 @@ int cfg_get_prplmesh_param_int(const std::string &param, int *buf)
+
+    utils::copy_string(ifname, interface_name.c_str(), BPL_IFNAME_LEN);
+    int status = cfg_uci_get_wireless_from_ifname(TYPE_RADIO, ifname, "channel", channel_num);
     if (status == RETURN_ERR)
```
This is silly - just change it to;
```
if (cfg_uci_get_wireless_from_ifname(TYPE_RADIO, ifname, "channel", channel_num) != RETURN_OK) {
    return RETURN_ERR;
}
```

adam1985d replied:
I agree - already fixed it locally - will push the commit after fixing all the comments

tomereli commented on framework/platform/bpl/uci/cfg/bpl_cfg_helper.cpp:
```diff
@@ -103,7 +106,7 @@ int cfg_get_channel(int index, int *channel)
 }
 
-int cfg_get_wep_key(int index, int keyIndex, char *key)
+int cfg_get_wep_key(const std::string &interface_name, int keyIndex, char *key)
```
I think this whole function can and should be removed. WEP is obsolete and we don't support it.

arnout replied:
But lets not make this PR block on that...

tomereli replied:
Right, forgot to mark this as (nitpick)

adam1985d replied:
ok - we can remove it as part of #805

tomereli commented on framework/platform/bpl/uci/cfg/bpl_cfg_uci.cpp:
```diff
@@ -66,12 +66,13 @@ int cfg_uci_get(char *path, char *value, size_t length)
     char val[MAX_UCI_BUF_LEN] = "";
 
-    status = cfg_uci_get_wireless(type, index, param, val);
+    status = cfg_uci_get_wireless_from_ifname(type, interface_name, param, val);
```
Similar here

arnout commented on framework/platform/bpl/uci/cfg/bpl_cfg_uci.cpp:
```diff
@@ -162,6 +162,132 @@ int cfg_uci_get_wireless(enum paramType type, int index, const char param[], cha
+        }
+    }
+
+    if (type == TYPE_RADIO) {
```
I really don't like the pattern
```
if (arg == 1) {
   ...
   return x;
}

// arg == 2
...
return y;
```

I.e., rely on the return for something that is a choice. The pattern is excellent for handling errors, but not for handling two alternative code paths IMO.

In this case, I much prefer either:
```
if (arg == 1) {
   ... foo = ...
} else { // arg == 2
   ... foo = ...
}
return foo;
```

or even

```
switch (arg) {
case 1:
   ...
   return x;
case 2:
   ...
   return y;
default:
   ERROR
   return z;
}
```

(nitpick)

tomereli replied:
Kudos on giving such an elaborate comment even though you mark it as (nitpick)

adam1985d replied:
the battle between the forces of indentation and forces of if-else continues :-)
I will change it to if-else

arnout commented on framework/platform/bpl/uci/cfg/bpl_cfg_uci.cpp:
```diff
@@ -162,6 +162,132 @@ int cfg_uci_get_wireless(enum paramType type, int index, const char param[], cha
+
+    if (!ptr.p) {
+        ERROR("%s, returned pointer is null\n", __FUNCTION__);
+        return RETURN_ERR;
```
Missing uci_free_context

adam1985d replied:
fixed

arnout commented on framework/platform/bpl/uci/cfg/bpl_cfg_uci.cpp:
```diff
@@ -162,6 +162,132 @@ int cfg_uci_get_wireless(enum paramType type, int index, const char param[], cha
+            struct uci_option *o = uci_to_option(n);
+
+            if (strncmp(n->name, "device", MAX_UCI_BUF_LEN) == 0 && o->type == UCI_TYPE_STRING) {
+                strncpy_s(deviceName, MAX_UCI_BUF_LEN, o->v.string, MAX_UCI_BUF_LEN - 1);
```
You could have constructed path here directly, but OK.

adam1985d replied:
changed to do exactly that. thanks

arnout commented on framework/platform/bpl/uci/cfg/bpl_cfg_uci.cpp:
```diff
@@ -162,6 +162,132 @@ int cfg_uci_get_wireless(enum paramType type, int index, const char param[], cha
+
+        char path[MAX_UCI_BUF_LEN] = "";
+        //get from wireless.radio?.param the param wanted
+        int status = snprintf_s(path, MAX_UCI_BUF_LEN, "wireless.%s.%s", deviceName, param);
```
Is there any particular reason to use snprintf_s rather than std::string?

adam1985d replied:
actually no - a bad example of copy-paste.
changed to use std::string

arnout commented on framework/platform/bpl/uci/cfg/bpl_cfg_uci.cpp:
```diff
@@ -162,6 +162,132 @@ int cfg_uci_get_wireless(enum paramType type, int index, const char param[], cha
+    if (type == TYPE_RADIO) {
+        //get radio name from .device param value
+        char deviceName[MAX_UCI_BUF_LEN] = {0};
+        uci_foreach_element(&s->options, n)
```
Is this still valid if is_section_found is false?

adam1985d replied:
you are correct - added a check after the loop. if not found - return false

arnout commented on framework/platform/bpl/uci/cfg/bpl_cfg_uci.cpp:
```diff
@@ -162,6 +162,132 @@ int cfg_uci_get_wireless(enum paramType type, int index, const char param[], cha
+    //interface not found in wireless
+    ERROR("%s, interface(%s) not found", __FUNCTION__, interface_name);
+    uci_free_context(ctx);
+    return RETURN_ERR;
```
So I think this bit should move up before the type == RADIO condition.

adam1985d replied:
done as a response to the previous comment. thanks

arnout commented on framework/platform/bpl/uci/cfg/bpl_cfg_uci.cpp:
```diff
@@ -162,6 +162,132 @@ int cfg_uci_get_wireless(enum paramType type, int index, const char param[], cha
+    // read the value from the selected section
+    if (is_section_found) {
+        // get param in the selected section
+        uci_foreach_element(&s->options, n)
+        {
+            struct uci_option *o = uci_to_option(n);
+            // if param is found in options
+            if (strncmp(n->name, param, MAX_UCI_BUF_LEN) == 0 && o->type == UCI_TYPE_STRING) {
```
Not specific for this commit, because it's done like this all over this file, but:

Why don't we use the uci helper functions? Like uci_parse_section, or for this specific one, uci_lookup_option?

adam1985d replied:
I was using the existing implementation as a reference and wasn't aware of the `uci_lookup_option`
since the access to wireless config is going to be removed in #805 I don't want to invest in refactoring it, but took notice for any later change in this area and in general. thanks

arnout commented on framework/platform/bpl/uci/cfg/bpl_cfg.cpp:
```diff
@@ -175,33 +175,30 @@ int cfg_get_device_info(BPL_DEVICE_INFO *device_info) { return 0; }
-        return retVal;
-    }
+    char ifname[BPL_IFNAME_LEN] = {0};
+    utils::copy_string(ifname, iface, BPL_IFNAME_LEN);
```
Why is this copied instead of just using `iface.c_str()`?

adam1985d replied:
The BPL UCI APIs provide a C interface.
although currently, we use it by CPP files only, @vitalybu suggested leaving the interface in C style for that purpose.
The BPL UCI APIs implementation requires non-const char arrays as the UCI lib indicates the provided strings might be changed inside and thus must not be const.

arnout replied:
Indeed, uci_lookup_ptr modifies its path argument. But it would be more appropriate to do the copy in uci_cfg_get() then. Anyway, something for a separate PR.

BTW, I believe cfg_uci_get_wireless() is no longer used and can be removed.

arnout commented on framework/platform/bpl/uci/cfg/bpl_cfg_uci.h:
```diff
@@ -91,6 +91,9 @@ int cfg_uci_get_radio_param(int index, const char param[], char *value, size_t b
 int cfg_uci_get_wireless_radio_idx(const char *interfaceName, int *radio_index);
 int cfg_uci_get_radio_param_ulong(int index, const char param[], unsigned long *value);
 
+int cfg_uci_get_wireless_from_ifname(enum paramType type, char *interface_name, const char param[],
```
interface_name and param should be const.

adam1985d replied:
changed to const

arnout commented on framework/platform/bpl/uci/cfg/bpl_cfg_helper.cpp:
```diff
@@ -82,14 +82,17 @@ int cfg_get_prplmesh_param_int(const std::string &param, int *buf)
+    char ifname[BPL_IFNAME_LEN]       = {0};
+
+    utils::copy_string(ifname, interface_name.c_str(), BPL_IFNAME_LEN);
+    int status = cfg_uci_get_wireless_from_ifname(TYPE_RADIO, ifname, "channel", channel_num);
```
Again, why copy ifname?

adam1985d replied:
removed once const is used for interface name

arnout commented on framework/platform/bpl/uci/cfg/bpl_cfg_uci.cpp:
```diff
@@ -66,12 +66,13 @@ int cfg_uci_get(char *path, char *value, size_t length)
 }
 
-int cfg_uci_get_wireless_int(enum paramType type, int index, const char param[], int *value)
+int cfg_uci_get_wireless_int(enum paramType type, char *interface_name, const char param[],
```
interface_name should be const

adam1985d replied:
fixed

arnout commented on framework/platform/bpl/uci/cfg/bpl_cfg_uci.cpp:
```diff
@@ -440,12 +441,12 @@ int cfg_uci_get_wireless_idx(char *interfaceName, int *rpc_index)
 }
 
-int cfg_uci_get_wireless_bool(enum paramType type, int index, const char param[], bool *value)
+int cfg_uci_get_wireless_bool(enum paramType type, char *interface_name, const char param[],
```
interface_name should be const

adam1985d replied:
fixed

arnout reviewed (approved):
Lots of changes requested, but can be merged after fixing those.

rmelotte commented on framework/platform/bpl/uci/cfg/bpl_cfg_uci.cpp:
```diff
@@ -162,6 +163,132 @@ int cfg_uci_get_wireless(enum paramType type, int index, const char param[], cha
+    struct uci_context *ctx = uci_alloc_context();
+    if (!ctx) {
+        ERROR("%s, uci alloc context failed!\n", __FUNCTION__);
+        return RETURN_ERR;
```
~~Shouldn't there also be a call to uci_free_context here?~~
Edit: nevermind

tomereli commented:
Added don't merge tag cause Adam needs to push some more code review fixes. @adam1985d please merge once you are done (and remove the tag).

adam1985d committed:
agent: platform: remove monitoring of unneeded wifi settings

Remove unneeded code.
The platform manager periodically reads the wifi configuration and on
detected change, sends the son_slave a notification with the new read
values. The son_slave ignores the SSID, passphrase, and security and
only checks the band_enabled value.
Remove monitoring of change in the SSID, passphrase, and security
settings.

Signed-off-by: Adam Dov <adamx.dov@intel.com>

adam1985d committed:
bpl: cfg_get_wifi_params: remove ssid, passphrase and security

Remove unneeded code.
The SSID, passphrase, and security are read from the wireless
configuration but are not used.
Remove read of the SSID, passphrase, and security from the wireless
config.

Signed-off-by: Adam Dov <adamx.dov@intel.com>

adam1985d committed:
bpl: add cfg_uci_get_wireless_from_ifname API

Preparative commit.
Add API to get a wireless param from the interface name.
This API removes the unnecessary dependency of BPL in the naming given
to the radio sections in the wireless UCI config.

Signed-off-by: Adam Dov <adamx.dov@intel.com>

adam1985d committed:
bpl: get wireless settings from the interface name

Change the BPL APIs used to read wireless configurations to use the new
cfg_uci_get_wireless_from_ifname API that is independent  of the radio
section names chosen for the wireless UCI configuration.

Signed-off-by: Adam Dov <adamx.dov@intel.com>
